crates/mock-download-server/src/server.rs

```diff
@@ -1,77 +1,97 @@
 // SPDX-FileCopyrightText: The Ferrocene Developers
 // SPDX-License-Identifier: MIT OR Apache-2.0
 
-use crate::handlers::handle_request;
-use crate::Data;
+use crate::{v1_routes, Data};
 use anyhow::Result;
+use axum::body::Body;
+use axum::extract::{Request, State};
+use axum::http::{Response, StatusCode};
+use axum::middleware::Next;
+use axum::response::IntoResponse;
+use axum::routing::get;
+use axum::{middleware, Router};
 use criticaltrust::manifests::{
     ManifestVersion, Package, PackageFile, PackageManifest, Release, ReleaseArtifact,
     ReleaseArtifactFormat, ReleaseManifest, ReleasePackage,
 };
 use criticaltrust::signatures::SignedPayload;
 use sha2::{Digest, Sha256};
 use std::fs::File;
+use std::net::{IpAddr, Ipv4Addr, SocketAddr};
 #[cfg(unix)]
 use std::os::unix::prelude::MetadataExt;
 #[cfg(windows)]
 use std::os::windows::prelude::MetadataExt;
 use std::path::Path;
-use std::sync::{Arc, Mutex, RwLock};
-use std::thread::JoinHandle;
-use tiny_http::Server;
+use std::sync::Arc;
+use tokio::net::TcpListener;
+use tokio::sync::{MappedMutexGuard, Mutex, MutexGuard};
+use tokio::task::JoinHandle;
 use walkdir::WalkDir;
 use xz2::write::XzEncoder;
 
 pub struct MockServer {
     data: Arc<Mutex<Data>>,
-    server: Arc<Server>,
+    address: SocketAddr,
     handle: Option<JoinHandle<()>>,
-    // This should be a `StatusCode` but they are not cloneable in `tiny_http`.
-    response_status_codes: Arc<RwLock<Vec<u16>>>,
 }
 
 impl MockServer {
-    pub(crate) fn spawn(data: Data) -> Self {
+    pub(crate) async fn spawn(data: Data) -> Self {
         let data = Arc::new(Mutex::new(data));
+        let data_clone = data.clone();
 
         // Binding on port 0 results in the operative system picking a random available port,
         // without the need of generating a random port ourselves and validating the port is not
         // being used by another process.
         //
         // The real port can be then retrieved by checking the address of the bound server.
-        let server = Arc::new(Server::http("127.0.0.1:0").unwrap());
-
-        let response_status_codes = Arc::new(RwLock::new(Vec::new()));
-
-        let data_clone = data.clone();
-        let server_clone = server.clone();
-        let response_status_codes_clone = response_status_codes.clone();
-        let handle = std::thread::spawn(move || {
-            server_thread(data_clone, server_clone, response_status_codes_clone);
+        let address = SocketAddr::new(IpAddr::V4(Ipv4Addr::new(127, 0, 0, 1)), 0);
+        let listener = TcpListener::bind(&address).await.unwrap();
+        let address = listener.local_addr().unwrap();
+
+        let mut this = Self {
+            data: data_clone,
+            address,
+            handle: None,
+        };
+        let router = Router::new()
+            .route("/", get(StatusCode::IM_A_TEAPOT))
```

Comment:
```suggestion
            .route("/", get(StatusCode::OK))
```
Sorry, this ain't a TEAPOT either ...

Reply:
This route is not tested, it's a TEAPOT to make it relatively clear it's not supported in the mock server. I suppose 501 would be more appropriate.